lib/modules/manager/nix/extract.ts

```diff
@@ -1,23 +1,129 @@
+import { logger } from '../../../logger';
 import { regEx } from '../../../util/regex';
 import { GitRefsDatasource } from '../../datasource/git-refs';
-import { id as nixpkgsVersioning } from '../../versioning/nixpkgs';
 import type { PackageDependency, PackageFileContent } from '../types';
+import { NixFlakeLock } from './schema';
 
-const nixpkgsRegex = regEx(/"github:nixos\/nixpkgs(\/(?<ref>[a-z0-9-.]+))?"/i);
+// TODO: add support to update nixpkgs branches in flakes.nix using nixpkgsVersioning
+
+// as documented upstream
+// https://github.com/NixOS/nix/blob/master/doc/manual/source/protocols/tarball-fetcher.md#gitea-and-forgejo-support
+const lockableHTTPTarballProtocol = regEx(
+  '^https://(.+)/(.+)/(.+)/archive/(.+).tar.gz$',
+);
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+): PackageFileContent | null {
+  logger.trace(`nix.extractPackageFile(${packageFile})`);
 
-export function extractPackageFile(content: string): PackageFileContent | null {
   const deps: PackageDependency[] = [];
 
-  const match = nixpkgsRegex.exec(content);
-  if (match?.groups) {
-    const { ref } = match.groups;
-    deps.push({
-      depName: 'nixpkgs',
-      currentValue: ref,
-      datasource: GitRefsDatasource.id,
-      packageName: 'https://github.com/NixOS/nixpkgs',
-      versioning: nixpkgsVersioning,
-    });
+  const flakeLockParsed = NixFlakeLock.safeParse(content);
```

Comment:
you need to search for lock file first, as this is the `flake.nix` content.

we've a getSilbling fs util function

Reply:
Although this comment is resolved, it doesn't seem to be implemented correctly -- the file path of the sibling is stored in the `packageLockFile` variable, but the contents that are parsed here are from `flake.nix` and not `flake.lock`.

The unit tests covering this in `extract.spec.ts` don't match the actual implementation as the packageFile is `flake.nix` and not `flake.lock`, which those unit tests are spec'd for.

I think that when https://github.com/renovatebot/renovate/pull/31921/commits/4bcf310a8d30e3f2eb7b8d12b886e6c6308249ba was reverted, all of this broke.